src/constfold.d

```diff
@@ -971,41 +971,127 @@ extern (C++) UnionExp Cmp(TOK op, Loc loc, Type type, Expression e1, Expression
             rawCmp = cast(int)(es1.len - es2.len);
         n = specificCmp(op, rawCmp);
     }
-    else if (e1.isConst() != 1 || e2.isConst() != 1)
-    {
-        emplaceExp!(CTFEExp)(&ue, TOKcantexp);
-        return ue;
-    }
-    else if (e1.type.isreal())
-    {
-        r1 = e1.toReal();
-        r2 = e2.toReal();
-        goto L1;
-    }
-    else if (e1.type.isimaginary())
-    {
-        r1 = e1.toImaginary();
-        r2 = e2.toImaginary();
-    L1:
-        n = realCmp(op, r1, r2);
-    }
-    else if (e1.type.iscomplex())
+    else if (e1.isConst() == 1 && e2.isConst() == 1)
     {
-        assert(0);
+        real_t r1, r2;
+        if (e1.type.isreal())
+        {
+            r1 = e1.toReal();
+            r2 = e2.toReal();
+            goto L1;
+        }
+        else if (e1.type.isimaginary())
+        {
+            r1 = e1.toImaginary();
+            r2 = e2.toImaginary();
+        L1:
+            n = realCmp(op, r1, r2);
+        }
+        else if (e1.type.iscomplex())
+        {
+            assert(0);
+        }
+        else
+        {
+            sinteger_t n1 = e1.toInteger(), n2 = e2.toInteger();
+            n = e1.type.isunsigned() || e2.type.isunsigned() ?
+                intUnsignedCmp(op, n1, n2) :
+                intSignedCmp(op, n1, n2);
+        }
     }
     else
```

Comment:
How about combining line 1002 with 1004 to make it `else if`? I think it would flow a little nicer.